src/api/lib/obsapi/markdown_renderer.rb

```diff
@@ -39,7 +39,11 @@ def link(link, title, content)
     end
 
     def block_code(code, language)
-      CodeRay.scan(code, language).div(css: :class)
+      if language
+        CodeRay.scan(code, language).div(css: :class)
+      else
+        "<pre><code>#{code}</code></pre>"
```

Comment:
Thanks for the fast patch!

This would not escape html characters, when there is no language detected. So scripts could be injected.

As described [here](https://github.com/vmg/redcarpet/issues/308#issuecomment-23780412), I would go for something like:

```ruby
def block_code(code, language)
  language ||= :plaintext
  CodeRay.scan(code, language).div(css: :class)
end
```

Reply:
The issue for me is that the difference between plaintext styling of `<pre><code><code></pre>` and coderay's plaintext `<div class="CodeRay"><div class="code"><pre></pre></div></div>`, which is a real waste in comparison.

Reply:
waste of what?

Reply:
I would really rather escape that manually than rely on coderay for this, just to skip the coderay for non-language codeblocks.